sqlserver/datadog_checks/sqlserver/metrics.py

```diff
@@ -987,8 +990,9 @@ def fetch_all_values(cls, cursor, counters_list, logger, databases=None):
         ctx = construct_use_statement(db)
         start = get_precise_time()
         try:
-            logger.debug("%s: changing cursor context via use statement: %s", cls.__name__, ctx)
-            cursor.execute(ctx)
+            if not is_azure_sql_database(engine_edition):
```

Comment:
I am trying to introduce a function like this - https://github.com/DataDog/integrations-core/pull/17258/files#r1579702387 
cause there are several places where we have this kind of logic, but fine for me to submit this PR as it is.